integration-testing/build.gradle.kts

```diff
@@ -0,0 +1,113 @@
+import org.jetbrains.kotlin.gradle.utils.NativeCompilerDownloader
```

Comment:
Is it possible to make integration tests module a submodule of the main project? Currently, it's not that convenient to work with integration tests and it's easy to forget about running them locally as tests declared here will not be executed along with other project tests.

Reply:
`integration-testing` is not included into `atomicfu` Gradle project, one have to manually `integration-testing` and run tests from it.

Reply:
I think in this case it is better to make a [composite build](https://docs.gradle.org/current/userguide/composite_builds.html) and run functional tests [when calling `check`](https://docs.gradle.org/current/userguide/composite_builds.html#included_build_task_dependencies) of the main project.

Reply:
We need this separate module to be able to run sample project builds completely independently, like they are honest isolated user projects 👀
And also, in integration-testing I use locally published current atomicfu version.

Reply:
But the sample projects will be independent either way as for them we invoke Gradle manually and these runs (or builds, I'm not sure about correct wording) will not depend on surrounding project (after all, all sample project files are copied into temp directory, aren't they?).

Reply:
Yes, you're right, project builds will be independent anyway. But I need to build them with a locally published atomicfu version, and I don't know how to make project tests depend on pubslishToMavenLocal task.

Reply:
Something like the following should work:
```
task checkAfterPublish(dependsOn: ':atomicfu-gradle-plugin:publishToMavenLocal')

tasks.named("check").configure {
    it.dependsOn(checkAfterPublish)
}
```